### PR TITLE
DAOS-4166 api: Remove API_VERSION from repository

### PR DESCRIPTION
**Pull request by jolivier23**
Generate the file, which is needed for raw python at the moment,
instead using SCons command builder.  This way we only have one
version number to modify in the top level SConstruct.

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Looks like: `$PREFIX/lib64/daos/API_VERSION` is not getting created, at least in the package builds:

```
[2021-02-08T16:58:31.382Z] error: File not found: /builddir/build/BUILDROOT/daos-1.1.2.1-9.5795.gd8cf39f3.el7.x86_64/usr/lib64/daos/API_VERSION
```

**Comment by jolivier23** [comment]
> Looks like: `$PREFIX/lib64/daos/API_VERSION` is not getting created, at least in the package builds:
> 
> ```
> [2021-02-08T16:58:31.382Z] error: File not found: /builddir/build/BUILDROOT/daos-1.1.2.1-9.5795.gd8cf39f3.el7.x86_64/usr/lib64/daos/API_VERSION
> ```

I can't figure out why?   It has me questioning whether it's even running the right version

**Review by brianjmurrell — APPROVED** [review]

File: SConstruct [hunk 1]
```diff
@@ -430,13 +427,16 @@ def scons(): # pylint: disable=too-many-locals
...
 
     env.Install('$PREFIX/etc/bash_completion.d', ['utils/completion/daos.bash'])
     env.Install('$PREFIX/lib/daos/TESTING/ftest/util',
                 ['utils/sl/env_modules.py'])
     env.Install('$PREFIX/lib/daos/TESTING/ftest/',
                 ['ftest.sh'])
+    api_version = env.Command("%s/API_VERSION" % build_prefix,
+                              "%s/SConstruct" % build_prefix,
+                              "echo %s > $TARGET" % (API_VERSION))
+    env.Install("$PREFIX/lib64/daos", api_version)
```
Comment (ashleypittman) [inline SConstruct:439]: Is this writing the file using echo, then installing the resulting "command" as a binary?  What's the file mode in this case?

**Review by ashleypittman — APPROVED** [review]
Approved, but I do wonder if there is a better option.

**Review by kccain — APPROVED** [review]

File: SConstruct (hunk 1 above)
Comment (jolivier23) [reply SConstruct:439]: It's not installing as a binary.  Install just copies files around.

jvolivie-desk1:[~/daos]$ ls install/lib64/daos/API_VERSION -la
-rw-r--r--. 1 jvolivie jvolivie 6 Feb  8 12:13 install/lib64/daos/API_VERSION

There probably is a better way to do it but I want to get the main problem resolved first (that there is more than one place to set the version).

**Pull request merged**